### PR TITLE
migrate button and withForwadedRef to typescript

### LINKED ISSUE #1061: Migrate Button component to typescript
## Purpose

Migrate Button component to typescript

### PR DESCRIPTION
**Pull request by igorbrasileiro**
#### What is the purpose of this pull request?
As the title says.
issue: #1061 
[Running workspace](https://brasbtn--storecomponents.myvtex.com/)

#### Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/ButtonPlain/index.tsx [hunk 1]
```diff
@@ -185,7 +185,11 @@ const ButtonPlain: FunctionComponent<Props> = props => {
       onMouseDown={props.onMouseDown}
       onFocus={props.onFocus}
       onBlur={props.onBlur}
-      ref={props.forwardedRef}
+      ref={
+        props.forwardedRef as React.RefObject<
```
Comment (matheusps) [inline react/components/ButtonPlain/index.tsx]: Destructure { RefObject }  from react.

File: react/components/Button/index.tsx [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+    onMouseUp,
+    onMouseDown,
+    onFocus,
+    onBlur,
+    forwardedRef,
+    type,
+    value,
+  } = props
+
+  React.useEffect(() => {
```
Comment (matheusps) [inline react/components/Button/index.tsx]: Destructure { useEffect } from react.

File: react/components/Button/index.tsx [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+  download: PropTypes.string,
+  /** When terciary, the upper case can be prevented */
+  noUpperCase: PropTypes.bool,
+  /** Disables label wrapping */
+  noWrap: PropTypes.bool,
+}
+
+type Props = InferProps<typeof propTypes>
+
+const Button: React.FC<Props> = props => {
```
Comment (matheusps) [inline react/components/Button/index.tsx]: Destructure { FC } from react.

File: react/components/Button/index.tsx (hunk 2 above)
Comment (emersonlaurentino) [reply react/components/Button/index.tsx]: @matheusps the proposed idea is to transform what we already have in javascript to typescript, not to use React's new features. Hooks is not available for all versions that use the styleguide within the context (vtexio). So improvements to the component should be made in the future, unfortunately.

**Comment by igorbrasileiro** [comment]
> @igorbrasileiro can u rabase with `features`?

Ok

File: react/components/Button/index.tsx (hunk 2 above)
Comment (matheusps) [reply react/components/Button/index.tsx]: Ow, okay. Got it.

Comment (matheusps) [reply react/components/Button/index.tsx]: I thought it was to improve the component itself, my bad.

**Pull request closed without merging**